### PR TITLE
Add --no-rhsm and LEAPP_NO_RHSM for skipping RHSM

### PR DESCRIPTION
**Pull request by drehak**
The behavior is the same as the legacy `LEAPP_DEVEL_SKIP_RHSM`, except supported.

Requries Repo PR: oamg/leapp-repository/pull/482

### CODE REVIEW

File: leapp/cli/upgrade/__init__.py [hunk 1]
```diff
@@ -156,10 +156,15 @@ def prepare_configuration(args):
     else:
         os.environ['LEAPP_EXPERIMENTAL'] = '0'
     os.environ['LEAPP_UNSUPPORTED'] = '0' if os.getenv('LEAPP_UNSUPPORTED', '0') == '0' else '1'
+    if args.no_rhsm or os.environ['LEAPP_NO_RHSM'] == '1':
```
Comment (vinzenz) [inline leapp/cli/upgrade/__init__.py]:
```suggestion
    if args.no_rhsm or os.getenv('LEAPP_NO_RHSM', '0') == '1':
```

File: leapp/cli/upgrade/__init__.py [hunk 2]
```diff
@@ -156,10 +156,15 @@ def prepare_configuration(args):
...
+        os.environ['LEAPP_NO_RHSM'] = '1'
+    else:
+        os.environ['LEAPP_NO_RHSM'] = os.environ.get('LEAPP_DEVEL_SKIP_RHSM', '0')
```
Comment (vinzenz) [inline leapp/cli/upgrade/__init__.py]:
```suggestion
        os.environ['LEAPP_NO_RHSM'] = os.getenv('LEAPP_DEVEL_SKIP_RHSM', '0')
```

File: leapp/cli/upgrade/__init__.py [hunk 3]
```diff
@@ -156,10 +156,15 @@ def prepare_configuration(args):
...
     configuration = {
         'debug': os.getenv('LEAPP_DEBUG', '0'),
         'verbose': os.getenv('LEAPP_VERBOSE', '0'),
         'whitelist_experimental': args.whitelist_experimental or ()
+        'no_rhsm': os.environ['LEAPP_NO_RHSM']
```
Comment (vinzenz) [inline leapp/cli/upgrade/__init__.py]:
```suggestion
        'no_rhsm': os.getenv('LEAPP_NO_RHSM', '0')
```

File: leapp/cli/upgrade/__init__.py (hunk 1 above)
Comment (drehak) [reply leapp/cli/upgrade/__init__.py]: Nice, thanks.

File: leapp/cli/upgrade/__init__.py (hunk 2 above)
Comment (drehak) [reply leapp/cli/upgrade/__init__.py]: Nice, thanks.

File: leapp/cli/upgrade/__init__.py (hunk 3 above)
Comment (drehak) [reply leapp/cli/upgrade/__init__.py]: In this case we certainly set `LEAPP_NO_RHSM` to some value, so I don't see the point in calling `getenv`.

Comment (vinzenz) [reply leapp/cli/upgrade/__init__.py]: true

File: leapp/cli/upgrade/__init__.py [hunk 4]
```diff
@@ -182,6 +185,7 @@ def process_whitelist_experimental(repositories, workflow, configuration, logger
 @command_opt('whitelist-experimental', action='append', metavar='ActorName', help='Enables experimental actors')
 @command_opt('debug', is_flag=True, help='Enable debug mode', inherit=False)
 @command_opt('verbose', is_flag=True, help='Enable verbose logging', inherit=False)
+@command_opt('no-rhsm', is_flag=True, help='Skip Red Hat Subscription Manager checks', inherit=False)
```
Comment (pirat89) [inline leapp/cli/upgrade/__init__.py]: It's something that customers could want to use and skipping checks is not exactly what about this is. What about something like:

```suggestion
@command_opt('no-rhsm', is_flag=True, help='Use only custom repositories and skip actions with Red Hat Subscription Manager', inherit=False)
```

You can modify it. Just take the point.

**Comment by pirat89** [comment]
@drehak probably broken, it seems the LEAPP_NO_RHSM is not set when --no-rhsm is used:

```
# LEAPP_NO_RHSM=1 leapp preupgrade --debug
...
============================================================
                           ERRORS                           
============================================================

2020-04-03 10:16:47.522338 [ERROR] Actor: verify_check_results
Message: RHSM is skipped but not custom repositories have been delivered.

============================================================
                       END OF ERRORS                        
============================================================

# leapp preupgrade --debug --no-rhsm
...
============================================================
                           ERRORS                           
============================================================

2020-04-03 10:12:46.338074 [ERROR] Actor: scan_subscription_manager_info
Message: A subscription-manager command failed to execute
Summary: 
    Details: Command ['subscription-manager', 'release'] failed with exit code 1.
    Stderr: This system is not yet registered. Try 'subscription-manager register --help' for more information.
    Hint: Please ensure you have a valid RHEL subscription and your network is up.

============================================================
                       END OF ERRORS                        
============================================================


```

**Comment by pirat89** [comment]
@drehak: also please bump the minor version of `framework_version` capability inside the spec file.

**Comment by AloisMahdal** [comment]
If leapp_upgrade_repositories.repo is empty, we ge this inhibitor: 

    Risk Factor: high (inhibitor)
    Title: Using RHSM has been skipped but no custom repositories have been delivered.
    Summary: Leapp is run in the mode when the Red Hat Subscription Manager is not used (the --no-rhsm option or the LEAPP_NO_RHSM=1 environment variable has been set) so leapp is not able to obtain YUM/DNF repositories with the content for the target system in the standard way. The content has to be delivered by the user manually.
    Remediation: [hint] Create the repository file according to instructions in the referred document on the following path with all repositories that should be used during the upgrade: "/etc/leapp/files/leapp_upgrade_repositories.repo".

The text mentions "the referred document" but does not refer any document.

How is cust going to know how to add the repos?  (Merely `cat`'ing may not work if they already  deployed the destination repos, because of duplicate id's.)

**Comment by pirat89** [comment]
Just waiting for finished testing of leapp-repository PR 482 as this needs to be merged together. However it's preapred for merge btw.

**Comment by bocekm** [comment]
@drehak, @pirat89, we need to add the new option and the env var to the man page.

**Pull request merged**